### PR TITLE
64 config

### LINKED ISSUE #64: project structure and common config file/s
We could collect the constants like texts, lists, paths, messages somewhere easy to find and access by users and devs. This makes it hopefully easier to change something without understanding the hole code structure of the project. I don't want to move everything into one file, but maybe group this stuff in a folder and think about file formats and names of the files. I don't have a solution and I don't think its required or urgent at all, but want to share some thoughts and open the discussion.

Right now we have a couple of files that are potentially related to this topic: 

```
resources/configuration.yml (only one line)
data/token
messages.yml
messages_de.yml  (shouldnt be in proj root)
resources/text_commands.csv  (not sure what its used for)
tests/conftest.py  (required by pytest not sure if we can rename it)
messages.py (loads messages.yml files)
config.py (collectection of project wide constants)
```

there are more files in proj root like `.gitignore, .dockerignore, Dockerfile, requirements.txt` but they are supposed to stay there.. 

For a chatbot text is a big thing. If you find a typo its way easier to find the message in a list than find the file and code snippet where its defined. However, it can be anoying as well to write code without knowing what the exact message is you are going to send. So we should try to use meaningful keys in the messages.yml. 

I'd also say configuration.yml and and text_commands.csv are not necessarily resources. Maybe we can split this stuff into: 
- text and alike thats tracked by git and somehow used by the bot and 
- static media that can be downloaded as zip (probably including the yml database but without scripts). 
I also like the idea of having a utility folder to throw in everything else :P.  All folders containing python modules should have an `__init__.py` file to make imports easier.

So with all that said my first suggestion (after discussion) would be add a config folder and move some of the files there (maybe folder names should be english for consistency):

- resources
- - database
- - rules
- - manoeverkarten
- config
- - messages.yml
- - messages_de.yml
- - settings.yml
- - __init__.py
- utility
- - convert_database.py
- - generate_resources.sh

Don't know what to do with the text_commands.csv. Are they generated dynamically based on the content on that file? Than I'd add them to config as well but maybe also as yaml?

### PR DESCRIPTION
**Pull request by lukruh**
Might be hard to see whats going on because so many files changed at once. I basically moved files around, changed import and loaded every yaml file in `config/__init__.py`. 

`text_commands.csv` -> `config/commands.yml`
`config.py` -> `config/constants.py`
`ressources/configuration.yml` -> `config/settings.yml` (this will become the main file for user configuration)

closes #64
depends on #65

### CODE REVIEW

**Comment by lukruh** [comment]
TL;DR: 
I'm not sure if the last commit is a good solution.. It would be nice to provide an option for custom settings files. Passing them as arguments makes it really easy to run multiple instances of the bot at the same time (different tokens) or just switch between debug and production mode (differnt log settings and maybe data paths...). 
Its generally good to be able to access the settings from everywhere, but we don't want to read the yaml files several times. Right now its easy to just import what we need wherever we need it, without passing the reference around. 

1. import base libraries
2. load config files
3. import other libraries that use messages

The downside of this approach is that the (hackish) `config.load(config, args)` needs to be called before some modules are imported. This is because loading the messages depends on the settings (language) and the settings loaded could depend on arguments. If this is a problem, we could remove arguments (maybe use env variables instead) or just create a config class load data when an instance is created and pass a reference to the instance around. Which I think is more anoying, than calling the load function before other imports.

1. import everything
2. create config object
3. pass config object to every function where its needed

UPDATE:
I was too sleepy, why not just parse the arguments directly in `config/__init__.py`... I overcomplicated it for no reason. PR will be updated soon

File: cogs/generalCog.py
Comment (XaverStiensmeier) [inline cogs/generalCog.py:1]:
```
  File "/home/xaver/Documents/Repos/pen_and_paper/ilaris/ilarisdiscordbot/cogs/generalCog.py", line 53, in card
    file_path = cfg.RESOURCES/"manoeverkarten"
NameError: name 'cfg' is not defined
```
must be changed everywhere.

**Review by XaverStiensmeier — CHANGES_REQUESTED** [review]
Check where `cfg` is still used and changes those occurrences.

File: cogs/generalCog.py
Comment (lukruh) [reply cogs/generalCog.py:1]: changed, I think this was the only one I missed(?)

**Review by XaverStiensmeier — APPROVED** [review]
